merkle-tree/indexed/tests/tests.rs

```diff
@@ -110,63 +113,281 @@ fn relayer_update<H>(
+    let onchain_tree: RefCell<
+        IndexedMerkleTree<H, MERKLE_TREE_HEIGHT, MERKLE_TREE_CHANGELOG, MERKLE_TREE_ROOTS>,
+    > = RefCell::new(IndexedMerkleTree::default());
+    onchain_tree.borrow_mut().init().unwrap();
+
+    // Local artifacts.
+    let mut local_indexing_array = IndexingArray::<H, INDEXING_ARRAY_ELEMENTS>::default();
+    let mut local_merkle_tree =
+        reference::IndexedMerkleTree::<H, MERKLE_TREE_HEIGHT, MERKLE_TREE_ROOTS>::new().unwrap();
+
+    // Insert a pair of nullifiers, correctly. Just do it with relayer.
+    let nullifier1: [u8; 32] = [
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 30,
+    ];
+    let nullifier2: [u8; 32] = [
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 10,
+    ];
+    onchain_queue.borrow_mut().append(nullifier1).unwrap();
+    onchain_queue.borrow_mut().append(nullifier2).unwrap();
+    let (new_low_nullifier, new_nullifier) = local_indexing_array.append(nullifier1).unwrap();
+    local_merkle_tree
+        .update(new_low_nullifier, new_nullifier)
         .unwrap();
+    let (new_low_nullifier, new_nullifier) = local_indexing_array.append(nullifier2).unwrap();
+    local_merkle_tree
+        .update(new_low_nullifier, new_nullifier)
+        .unwrap();
+    relayer_update(
+        &mut onchain_queue.borrow_mut(),
+        &mut onchain_tree.borrow_mut(),
+    )
+    .unwrap();
+
+    // Try inserting nullifier 20, while pointing to index 1 (value 30) as low
+    // nullifier. Point to index 2 (value 10) as next value.
+    // Therefore, the new element is lowe than the supposed low element.
+    let nullifier3: [u8; 32] = [
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 20,
+    ];
+    onchain_queue.borrow_mut().append(nullifier3).unwrap();
+    let changelog_index = onchain_tree.borrow_mut().changelog_index();
+    // Index of our new nullifier in the queue.
+    let queue_index = 1_u16;
+    // Index of our new nullifier in the tree / on-chain state.
+    let nullifier_index = 3_u16;
+    // (Invalid) index of the next nullifier.
+    let nullifier_next_index = 2_u16;
+    // (Invalid) value of the next nullifier.
+    let nullifier_next_value = nullifier2;
+    // (Invalid) low nullifier.
+    let low_nullifier = local_indexing_array.get(1).cloned().unwrap();
+    let low_nullifier_proof = local_merkle_tree.get_proof_of_leaf(1);
+    assert!(matches!(
+        program_update(
+            &mut onchain_queue.borrow_mut(),
+            &mut onchain_tree.borrow_mut(),
+            changelog_index as u16,
+            queue_index,
+            nullifier_index,
+            nullifier_next_index,
+            nullifier_next_value,
+            low_nullifier,
+            low_nullifier_proof,
+        ),
+        Err(HasherError::LowElementGreaterOrEqualToNewElement)
+    ));
```

Comment:
we can be sure that the state of the queue and the tree didn't change right?

Reply:
Yes